pulser-core/pulser/register/base_register.py

```diff
@@ -233,7 +242,7 @@ def __eq__(self, other: Any) -> bool:
         if type(other) is not type(self):
             return False
 
-        return set(self._ids) == set(other._ids) and all(
+        return list(self._ids) == list(other._ids) and all(
```

Comment:
I'm not sure about this. What was wrong with the old way?

Reply:
The old way didn't check the ids order.

Reply:
But that's ok, perhaps I expressed myself poorly in my previous suggestion. What I meant is the the ids should still match the same coordinates after encoding-decoding

Reply:
Actually that's quite not okay, because `find_indices` relies on the order of the ids. The coordinates are already checked in the `__eq__` function.

Reply:
Yes, you're right 🙂